lib/internal/streams/operators.js

```diff
@@ -0,0 +1,269 @@
+'use strict';
+
+const { AbortController } = require('internal/abort_controller');
+const { isIterable } = require('internal/streams/utils');
+const {
+  codes: {
+    ERR_INVALID_ARG_TYPE,
+    ERR_INVALID_ARG_VALUE,
+  }, AbortError,
+} = require('internal/errors');
+
+const {
+  NumberIsInteger,
+  Promise,
+  PromiseReject,
+  Symbol,
+} = primordials;
+
+const kEmpty = Symbol('kEmpty');
+
+async function * map(fn, options) {
+  if (typeof fn !== 'function') {
+    throw new ERR_INVALID_ARG_TYPE(
+      'fn', ['Function', 'AsyncFunction'], this);
+  }
+
+  let concurrency = 1;
+  if (options?.concurrency != null) {
+    concurrency = options.concurrency;
+  } else if (options != null && typeof options !== 'object') {
+    concurrency = options;
+  }
+
+  if (!NumberIsInteger(concurrency)) {
+    throw new ERR_INVALID_ARG_TYPE(
+      'concurrency', ['Integer'], concurrency);
+  }
+
+  if (concurrency < 1) {
+    throw new ERR_INVALID_ARG_VALUE('concurrency', concurrency);
+  }
+
+  const ac = new AbortController();
+  const stream = this;
+  const queue = [];
+  const signal = ac.signal;
+  const signalOpt = { signal };
+
+  options?.signal?.addEventListener('abort', () => ac.abort(), { weak: true });
```

Comment:
That's not how you make weak event handlers :)

You do `[kWeakHandler]: <whateverIsRetainingTheListener>`.